x/ccv/provider/keeper/msg_server.go

```diff
@@ -293,32 +313,342 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(types.AttributeConsumerId, msg.ConsumerId),
+			sdk.NewAttribute(types.AttributeConsumerChainId, chainId),
 			sdk.NewAttribute(types.AttributeProviderValidatorAddress, msg.ProviderAddr),
 			sdk.NewAttribute(types.AttributeConsumerCommissionRate, msg.Rate.String()),
+			sdk.NewAttribute(types.AttributeSubmitterAddress, msg.Submitter),
 		),
-	})
+	)
 
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// CreateConsumer creates a consumer chain
+func (k msgServer) CreateConsumer(goCtx context.Context, msg *types.MsgCreateConsumer) (*types.MsgCreateConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	resp := types.MsgCreateConsumerResponse{}
+
+	// initialize an empty slice to store event attributes
+	eventAttributes := []sdk.Attribute{}
+
+	consumerId := k.Keeper.FetchAndIncrementConsumerId(ctx)
+
+	k.Keeper.SetConsumerOwnerAddress(ctx, consumerId, msg.Submitter)
+	k.Keeper.SetConsumerChainId(ctx, consumerId, msg.ChainId)
+	k.Keeper.SetConsumerPhase(ctx, consumerId, types.CONSUMER_PHASE_REGISTERED)
+
+	if err := k.Keeper.SetConsumerMetadata(ctx, consumerId, msg.Metadata); err != nil {
+		return &resp, errorsmod.Wrapf(types.ErrInvalidConsumerMetadata,
+			"cannot set consumer metadata: %s", err.Error())
+	}
+
+	// add event attributes
+	eventAttributes = append(eventAttributes, []sdk.Attribute{
+		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+		sdk.NewAttribute(types.AttributeConsumerId, consumerId),
+		sdk.NewAttribute(types.AttributeConsumerChainId, msg.ChainId),
+		sdk.NewAttribute(types.AttributeConsumerName, msg.Metadata.Name),
+		sdk.NewAttribute(types.AttributeSubmitterAddress, msg.Submitter),
+		sdk.NewAttribute(types.AttributeConsumerOwner, msg.Submitter),
+	}...)
+
+	// initialization parameters are optional and hence could be nil;
+	// in that case, set the default
+	initializationParameters := types.ConsumerInitializationParameters{} // default params
+	if msg.InitializationParameters != nil {
+		initializationParameters = *msg.InitializationParameters
+	}
+	if err := k.Keeper.SetConsumerInitializationParameters(ctx, consumerId, initializationParameters); err != nil {
+		return &resp, errorsmod.Wrapf(types.ErrInvalidConsumerInitializationParameters,
+			"cannot set consumer initialization parameters: %s", err.Error())
+	}
+	if !initializationParameters.SpawnTime.IsZero() {
+		// add SpawnTime event attribute
+		eventAttributes = append(eventAttributes,
+			sdk.NewAttribute(types.AttributeConsumerSpawnTime, initializationParameters.SpawnTime.String()))
+	}
```

Comment:
Feels like this should go in line 409 after the If on InitializeConsumer and use the SpawnTime from there